### PR TITLE
Non-linear mapping, part0

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/sub.mk [hunk 1]
```diff
@@ -33,7 +33,7 @@ srcs-y += wait_queue.c
 srcs-$(CFG_PM_STUBS) += pm_stubs.c
 
 srcs-$(CFG_GENERIC_BOOT) += generic_boot.c
-srcs-$(CFG_GENERIC_BOOT) += generic_core_bootcfg.c
+srcs-y += generic_core_bootcfg.c
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/kernel/sub.mk]: Should we remove `generic` from the filename, and rename it as `core_bootcfg_mm.c` ?

**Comment by pascal-brand-st-dev** [comment]
`Tested-by: Pascal Brand <pascal.brand@linaro.org> (STM)`

@jforissier Could you review it as you already played with it?

File: core/arch/arm/mm/tee_mmu.c [hunk 2]
```diff
@@ -629,21 +629,27 @@ void teecore_init_ta_ram(void)
...
 	core_mmu_get_mem_by_type(MEM_AREA_TA_RAM, &s, &e);
+	ps = virt_to_phys((void *)s);
+	TEE_ASSERT(ps);
+	pe = virt_to_phys((void *)(e - 1)) + 1;
+	TEE_ASSERT(pe);
 
-	TEE_ASSERT((s & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
-	TEE_ASSERT((e & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
+	TEE_ASSERT((ps & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
+	TEE_ASSERT((pe & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
```
Comment (jforissier) [inline core/arch/arm/mm/tee_mmu.c]: I don't understand these two lines.

File: core/tee/tee_svc.c [hunk 3]
```diff
@@ -658,7 +658,7 @@ static TEE_Result tee_svc_copy_param(struct tee_ta_session *sess,
 	dst_pa = tee_mm_get_smem(*mm);
 
 	/* Get the virtual address for the section in secure DDR */
-	dst = phys_to_virt(dst_pa, MEM_AREA_TEE_RAM);
+	dst = phys_to_virt(dst_pa, MEM_AREA_TA_RAM);
```
Comment (jforissier) [inline core/tee/tee_svc.c]: The commit comment does not really reflect this change, does it?

Comment (jenswi-linaro) [inline core/tee/tee_svc.c]: You're right, I'll update the commit message.

File: core/arch/arm/mm/tee_mmu.c (hunk 2 above)
Comment (jenswi-linaro) [inline core/arch/arm/mm/tee_mmu.c]: They are checking the the start and end are properly aligned. Not really needed in my opinion, but I was fixing the va and pa mixup in this patch.

Comment (jforissier) [inline core/arch/arm/mm/tee_mmu.c]: OK that's fine

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/kernel/sub.mk (hunk 1 above)
Comment (jenswi-linaro) [inline core/arch/arm/kernel/sub.mk]: This file is removed (content moved into `core/arch/arm/mm/core_mmu.c`) in a later patch from https://github.com/jenswi-linaro/optee_os/tree/nlin_map

**Comment by pascal-brand-st-dev** [comment]
`Reviewed-by: Pascal Brand <pascal.brand@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied

**Pull request merged**